pdk/1.0/modules/ROOT/pages/policies-pdk-configure-features-dataweave.adoc

```diff
@@ -0,0 +1,216 @@
+= Using DataWeave Expressions
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+:imagesdir: ../assets/images
+
+NOTE: To view an example policy project that uses DataWeave expressions, see https://github.com/mulesoft/pdk-custom-policy-examples/blob/main/query/README.md[Query Policy Example^].
+
+If the input DataWeave expression parameter is valid, the policy parses the expression and transforms it into a `pdk::script::Script` expression. To obtain the result of the script:
+
+. Create an evaluator.
+. Call the binding function for each parameter defined in the schema definition:
++
+* `vars`: Call the `bind_vars` method for each of the policy's `vars`. After all of `vars` are bound, the evaluator tries to solve the expression. If the expression contains `vars` other than the ones defined in the schema, they resolve to `null`.
+* `attributes`: Call the `bind_attributes` method with an implementation of the `AttributesBinding` trait. PDK provides an implementation of this trait by instantiating a new `HandlerAttributesBinding` for each `RequestHeaderState`, `RequestHeaderState`, and response of an xref:policies-pdk-configure-features-http-request.adoc[HTTP call].
+* `authentication`: Call the `bind_authentication` method with an implementation of the trait `AuthenticationBinding`. PDK provides an implementation of this trait for `AuthenticationData`. To inject authentication information into your policy, see xref:policies-pdk-configure-features-inject-parameters.adoc#authentication[Inject Authentication Inforamaion].
+* `payload`: Call the `bind_payload` method with an implementation of the trait `PayloadBinding`. PDK provides an implementation of this trait for `RequestBodyState`, `ResponseBodyState`, and the response of an xref:policies-pdk-configure-features-http-request.adoc[HTTP call].
++
+[NOTE]
+====
+Call only the methods for the bindings defined in the schema definition. For example, if the `attributes` binding is not used, don't call the `bind_attributes` method.
+
+To learn more about defining DataWeave expressions, see xref:policies-pdk-create-schema-definition.adoc#dataweave-expressions[DataWeave Expressions].
+====
+
+. After you bind all expression variables, obtain the result by calling the `eval` method.
+
+[[example-evaluator-code]]
+== Example Evaluator Code
+
+For example, to evaluate the DataWeave expression in the following policy schema definition:
```

Comment:
Which expression? "#[vars.myVar]"? I assume so, but I'm not sure at first read. Also what do they have to do to evaluate the expression? Include the expression as the value of `default`?  And what is `default`? Is this explained somewhere? Again, as a newbie I'm not sure what all this means. Will the average customer already know about this stuff?

Reply:
https://github.com/mulesoft/docs-gateway/pull/504/files